roda-core/roda-core/src/main/java/org/roda/core/plugins/base/preservation/EditFileFormatPlugin.java

```diff
@@ -0,0 +1,302 @@
+import org.roda.core.data.v2.LiteOptionalWithCause;
+import org.roda.core.data.v2.ip.File;
+import org.roda.core.data.v2.ip.StoragePath;
+import org.roda.core.data.v2.ip.metadata.LinkingIdentifier;
+import org.roda.core.data.v2.jobs.Job;
+import org.roda.core.data.v2.jobs.PluginParameter;
+import org.roda.core.data.v2.jobs.PluginState;
+import org.roda.core.data.v2.jobs.PluginType;
+import org.roda.core.data.v2.jobs.Report;
+import org.roda.core.data.v2.validation.ValidationException;
+import org.roda.core.index.IndexService;
+import org.roda.core.model.ModelService;
+import org.roda.core.model.utils.ModelUtils;
+import org.roda.core.plugins.AbstractPlugin;
+import org.roda.core.plugins.Plugin;
+import org.roda.core.plugins.PluginException;
+import org.roda.core.plugins.PluginHelper;
+import org.roda.core.plugins.RODAObjectsProcessingLogic;
+import org.roda.core.plugins.orchestrate.JobPluginInfo;
+import org.roda.core.storage.DirectResourceAccess;
+import org.roda.core.storage.StorageService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Alexandre Flores <aflores@keep.pt>
+ */
+public class EditFileFormatPlugin extends AbstractPlugin<File> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(EditFileFormatPlugin.class);
+  private static Map<String, PluginParameter> pluginParameters = new HashMap<>();
+
+  static {
+    pluginParameters.put(RodaConstants.PLUGIN_PARAMS_MIMETYPE,
+      PluginParameter
+        .getBuilder(RodaConstants.PLUGIN_PARAMS_MIMETYPE, "Mimetype", PluginParameter.PluginParameterType.STRING)
+        .withDescription("The Mimetype to set for all selected files.").build());
+    pluginParameters.put(RodaConstants.PLUGIN_PARAMS_FORMAT,
+      PluginParameter
+        .getBuilder(RodaConstants.PLUGIN_PARAMS_FORMAT, "Format", PluginParameter.PluginParameterType.STRING)
+        .withDescription("The full format descriptor to set for all selected files.").build());
+    pluginParameters.put(RodaConstants.PLUGIN_PARAMS_FORMAT_VERSION,
+      PluginParameter
+        .getBuilder(RodaConstants.PLUGIN_PARAMS_FORMAT_VERSION, "Format version",
+          PluginParameter.PluginParameterType.STRING)
+        .withDescription("The version for the format descriptor to set for all selected files.").build());
+    pluginParameters.put(RodaConstants.PLUGIN_PARAMS_PRONOM,
+      PluginParameter
+        .getBuilder(RodaConstants.PLUGIN_PARAMS_PRONOM, "PRONOM", PluginParameter.PluginParameterType.STRING)
+        .withDescription("The PRONOM identifier to set for all selected files.").build());
+  }
+
+  private String mimetype;
+  private String format;
+  private String formatVersion;
+  private String pronom;
+
+  public static String getStaticName() {
+    return "Edit File Format";
+  }
+
+  public static String getStaticDescription() {
+    return "Overwrites the selected files' extension, mimetype, format and PRONOM identifier with provided values.";
```

Comment:
```suggestion
    return "Overwrites the selected files' extension, MIME type, format and PRONOM identifier with provided values.";
```

Reply:
Resolved in https://github.com/keeps/roda/pull/3255/commits/de2e94ce3f2ffe9099904d69b8e10203caf3a2fb